File: src/problem039/solution039.py
import math

def main():
    perimeter_limit = 1000
    perimeter_solns = {}
    for a in xrange(1, perimeter_limit):
        for b in xrange(1, perimeter_limit - a):
            c = math.sqrt(a ** 2 + b ** 2)
            perimeter = int(a + b + c)
            if c == int(c) and a + b + c <= perimeter_limit:
                perimeter_solns[perimeter] = perimeter_solns.get(perimeter, 0) + 1
    perimeter, count = max(perimeter_solns.items(), key=lambda (k, v): v)
    print(perimeter)

if __name__ == '__main__':
    main()

File: src/problem039/solution039.js
var perims = {};
var limit = 1000;

for (var a = 1; a < limit; a++) {
    for (var b = 1; b <= a; b++) {
        var c = Math.sqrt(a * a + b * b);
        var p = parseInt(a + b + c);
        if (c == parseInt(c) && p <= limit) {
            perims[p] = 1 + (perims[p] || 0);
        }
    }
}
var max;
for (var p in perims) {
    if (!max || perims[p] > perims[max])
        max = p;
}
console.log(max);